logger.go

```diff
@@ -0,0 +1,175 @@
+package glog
+
+import (
+	"context"
+	"fmt"
+)
+
+// Logger provides logging functionality with additional data and prefixing.
+type Logger struct {
+	*loggingT
+	prefix string
+	// Extra arguments to be appended to each request
+	data []interface{}
+}
+
+// NewLogger creates a Logger instance with no additional data.
+func NewLogger() Logger {
```

Comment:
The pointer vs. value discussion. This seems to be a good candidate for a pointer.

Reply:
Possibly, it just makes the copying for modification more difficult. Looking at the structure of context.Context, it should be relatively straightforward to use as inspiration.

Reply:
Converted to a pointer type, and created a fresh Logger in full. The reason we want a new one is so we can pass logger.WithData() down into a loop and then still have the original set of data afterwards.

Reply:
(is there a better way / convention to dismiss comments here? other than using emoji; ideally, I would like them to disappear from my view, but not get deleted entirely)

Reply:
I think it marks things as outdated when the code in question changes. Not the best fit for the way we've usually done reviews.